src/xcode/ENA/ENA/Source/Services/TicketValidation/TicketValidation.swift

```diff
@@ -43,5 +44,92 @@ final class TicketValidation: TicketValidating {
 	func cancel() {
 
 	}
-
+	
+	func encryptAndSignDCC(
+		dccBarcodeData: String,
+		nonceBase64: String,
+		encryptionScheme: EncryptionScheme,
+		publicKeyForEncryption: SecKey,
+		privateKeyForSigning: SecKey
+	) -> Result<EncryptAndSignResult, EncryptAndSignError> {
+		
+		// Determine iv (initializationVector): the iv shall be derived as the byte sequence (Data / ByteArray) of nonceBase64 using base64 encoding.
+		guard let initializationVector = Data(base64Encoded: nonceBase64) else {
+			return .failure(.UNKNOWN)
```

Comment:
Maybe add a Log statement for each return what step failed? Could make it easier to debug.

Reply:
Added logs.